docker-compose/compose.yml

```diff
@@ -0,0 +1,418 @@
+secrets:
+  mysql_password:
+    file: ./secrets/mysql_password.txt
+  mysql_root_password:
+    file: ./secrets/mysql_root_password.txt
+
+services:
+  mysql:
+    image: mysql:${MYSQL_VERSION:-8.0}
+    container_name: bucketeer-mysql
+    restart: unless-stopped
+    environment:
+      MYSQL_ROOT_PASSWORD_FILE: /run/secrets/mysql_root_password
+      MYSQL_DATABASE: bucketeer
+      MYSQL_USER: bucketeer
+      MYSQL_PASSWORD_FILE: /run/secrets/mysql_password
+    secrets:
+      - mysql_password
+      - mysql_root_password
+    ports:
+      - "3306:3306"
+    volumes:
+      - mysql_data:/var/lib/mysql
+      - ./init-db:/docker-entrypoint-initdb.d
+    command: --default-authentication-plugin=mysql_native_password --character-set-server=utf8mb4 --collation-server=utf8mb4_general_ci
+    healthcheck:
+      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "bucketeer", "-pbucketeer"]
+      timeout: 20s
+      retries: 10
+    networks:
+      - bucketeer
+
+  redis:
+    image: redis:${REDIS_VERSION:-7-alpine}
+    container_name: bucketeer-redis
+    restart: unless-stopped
+    ports:
+      - "6379:6379"
+    volumes:
+      - redis_data:/data
+    command: redis-server --appendonly yes --appendfsync everysec --no-appendfsync-on-rewrite no --auto-aof-rewrite-percentage 100 --auto-aof-rewrite-min-size 64mb --save 900 1 --save 300 10 --save 60 10000
```

Comment:
Since Redis stores Redis Streams data (cache, event data, metrics, audit logs, etc.), the previous configuration had minimal persistence settings and had a risk of permanent data loss if Redis runs out of memory or crashes.